csm/core/services/health.py

```diff
@@ -58,8 +58,32 @@ def __init__(self, repo: HealthRepository, alerts_repo, plugin):
         self.alerts_repo = alerts_repo
         self._is_map_updated_with_db = False
         self._health_plugin = plugin
+        self._node_hostname_map = dict()
+        self._hostname_node_map = dict()
+        self._create_node_hostname_map()
         self._init_health_schema()
 
+    def _create_node_hostname_map(self):
+        """
+        This method creates an in-memory map of minion-id to hostname and
+        vice-versa.
+        """
+        self._node_hostname_map = Conf.get(const.CSM_GLOBAL_INDEX, f"{const.MAINTENANCE}")
+        self._node_hostname_map.pop(const.SHUTDOWN_CRON_TIME)
+        self._hostname_node_map = {host:node for node, host in self._node_hostname_map.items()}
+
+    def get_minion_id(self, hostname):
+        """
+        Returns minion id for the corresponding hostname.
+        """
+        return self._hostname_node_map.get(hostname, "")
```

Comment:
Throw an exception if hostname is not found.

Reply:
Done